### PR TITLE
Fixed PSMA indicator not flagging port visits to non-PSMA port states. VV-512

### PR DESCRIPTION
**Pull request by rdgfuentes**
https://globalfishingwatch.atlassian.net/browse/VV-512

### CODE REVIEW

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts [hunk 1]
```diff
@@ -2,79 +2,80 @@ import { createSelector } from '@reduxjs/toolkit'
...
+export const getEventsWithMainPortVisit = (events): RenderedEvent[] =>
+  events.reduce((previous, event) => {
+    const isPortExit = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Exit
+    const isPortEntry = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Entry
+
+    if (isPortExit) {
+      // If it's a port exit then build the single port visit event from it
+      const id = event.id.split('-').shift()
+      const portEntry =
+        (isPortEntry && event) || events.find((ev) => ev.id === `${id}-${PortVisitSubEvent.Entry}`)
```
Comment (j8seangel) [inline apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:20]: can an event be portExit and portEntry at the same time? if not I think this `(isPortEntry && event)` is not needed
also, could be possible to have portExit without portEntry ?

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts [hunk 2]
```diff
@@ -2,79 +2,80 @@ import { createSelector } from '@reduxjs/toolkit'
...
-      const isPortExit = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Exit
-      const isPortEntry =
-        event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Entry
+export const getEventsWithMainPortVisit = (events): RenderedEvent[] =>
+  events.reduce((previous, event) => {
+    const isPortExit = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Exit
+    const isPortEntry = event.type === EventTypes.Port && event.subEvent === PortVisitSubEvent.Entry
+
+    if (isPortExit) {
+      // If it's a port exit then build the single port visit event from it
```
Comment (j8seangel) [inline apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:17]: thanks for the comment, helped a lot to understand the context

**Review by j8seangel — COMMENTED** [review]
Just some technical comments, in general LGTM

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts (hunk 1 above)
Comment (rdgfuentes) [reply apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts]: Good point! Based on that I noticed and scenario (having only port visit entry event) where we would not have shown the main port visit so I made some adjustments to fix this problem.

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts [hunk 3]
```diff
@@ -43,11 +41,23 @@ export const getEventsWithMainPortVisit = (events): RenderedEvent[] =>
...
+        ? Duration.fromMillis(portVisitEvent.port_visit?.durationHrs * 60 * 60 * 1000)
+            .shiftTo('hours', 'minutes')
+            .toObject()
+        : portExit?.end > portEntry?.start
+        ? DateTime.fromMillis(portExit?.end as number)
+            .diff(DateTime.fromMillis(portEntry?.start as number), ['hours', 'minutes'])
+            .toObject()
+        : null
+
+      console.log(duration)
```
Comment (j8seangel) [inline apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:54]: 👀 ⚠️

**Review by j8seangel — APPROVED** [review]
LGTM

File: apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts (hunk 3 above)
Comment (rdgfuentes) [reply apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts:54]: 🧹

**Pull request merged**